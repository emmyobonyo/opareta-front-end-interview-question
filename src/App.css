* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.container {
  width: 50%;
  margin: auto;
  margin-top: 50px;
  padding: 15px;
  background-color: rgb(240, 240, 240);
  border-radius: 5px;
}

.input {
  width: 45%;
  padding: 5px;
  margin-bottom: 10px;
  border: 1px solid rgb(202, 202, 202);
}

.select {
  width: 45%;
  padding: 5px;
  border: 1px solid rgb(202, 202, 202);
}

button {
  padding: 5px;
  margin: 0 17px;
  background-color: lightskyblue;
  border: none;
  color: white;
}

p {
  text-align: center;
  font-weight: bold;
  font-family: Arial, Helvetica, sans-serif;
  margin-top: 10px;
  font-size: 40px;
}
