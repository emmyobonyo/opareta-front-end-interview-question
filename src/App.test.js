import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

test('should return the home page', () => {
  const home = render(<App />);
  expect(home).toMatchSnapshot();
  expect(home.queryByText(/Bitcoin/)).toBeInTheDocument();
})

// I believe the test below is important because it ensures the output value exists.
// Without the output value, the imputs are useless

test('should return the home page', () => {
  const home = render(<App />);
  expect(home.queryByText(/0/)).toBeInTheDocument();
})

