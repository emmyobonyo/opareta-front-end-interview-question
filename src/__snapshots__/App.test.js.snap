// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="container"
        >
          <input
            class="input"
            id="number"
            placeholder="Enter Amount"
            type="number"
          />
          <br />
          <select
            class="select"
            id="crypto"
            name="crypto"
          >
            <option
              value="1"
            >
              Bitcoin (BTC)
            </option>
            <option
              value="2"
            >
              Litecoin (LIT)
            </option>
            <option
              value="3"
            >
              Namecoin (NMC)
            </option>
          </select>
          <button
            type="button"
          >
            <i
              aria-hidden="true"
              class="fa fa-exchange"
            />
          </button>
          <select
            class="select"
            id="currency"
            name="currency"
          >
            <option
              value="USD"
            >
              US Dollars ($)
            </option>
            <option
              value="GBP"
            >
              Pound Sterling (£)
            </option>
            <option
              value="UGX"
            >
              Uganda Shillings (/-)
            </option>
          </select>
          <p>
            0
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="container"
      >
        <input
          class="input"
          id="number"
          placeholder="Enter Amount"
          type="number"
        />
        <br />
        <select
          class="select"
          id="crypto"
          name="crypto"
        >
          <option
            value="1"
          >
            Bitcoin (BTC)
          </option>
          <option
            value="2"
          >
            Litecoin (LIT)
          </option>
          <option
            value="3"
          >
            Namecoin (NMC)
          </option>
        </select>
        <button
          type="button"
        >
          <i
            aria-hidden="true"
            class="fa fa-exchange"
          />
        </button>
        <select
          class="select"
          id="currency"
          name="currency"
        >
          <option
            value="USD"
          >
            US Dollars ($)
          </option>
          <option
            value="GBP"
          >
            Pound Sterling (£)
          </option>
          <option
            value="UGX"
          >
            Uganda Shillings (/-)
          </option>
        </select>
        <p>
          0
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
